toolchain/check/pattern_match.cpp

```diff
@@ -263,9 +261,7 @@ auto MatchContext::DoEmitPatternMatch(Context& context,
     default:
       CARBON_DIAGNOSTIC(AddrSelfIsNonRef, Error,
                         "`addr self` method cannot be invoked on a value");
-      context.emitter().Emit(
-          TokenOnly(context.insts().GetLocId(entry.scrutinee_id)),
-          AddrSelfIsNonRef);
+      context.emitter().Emit(entry.scrutinee_id, AddrSelfIsNonRef);
```

Comment:
This was previously `TokenOnly`

Reply:
Yeah, this was intentional, see here: https://github.com/carbon-language/carbon-lang/pull/5355#discussion_r2060406639

Reply:
Can this use `GetCanonical`, and let geoffromer review if the diagnostic was incorrect? I think it shouldn't change -- this PR should be refactoring only.

Reply:
Note, to be sure -- my suspicion is that this is a regression.

Reply:
Maybe as a TODO, had you considered moving `TokenOnly` tracking to the added `LocIdForDiagnostics` (perhaps keeping `auto TokenOnly(LocIdForDiagnostics) -> LocIdForDiagnostics`), removing it from `LocId`?

Reply:
Re TODO: That's an interesting thought.

Re GetCanonical: Yes, sure, will do.

Reply:
Done